lib/AdobeState.js

```diff
@@ -317,20 +320,31 @@ class AdobeState {
         },
         value: {
           type: 'string'
+        },
+        ttl: {
+          type: 'integer'
         }
       }
     }
 
-    const { valid, errors } = validate(schema, { key, value })
+    // validation
+    const { ttl } = options
+    const { valid, errors } = validate(schema, { key, value, ttl })
     if (!valid) {
       logAndThrow(new codes.ERROR_BAD_ARGUMENT({
         messageValues: utils.formatAjvErrors(errors),
-        sdkDetails: { key, value, options, errors }
+        sdkDetails: { key, valueLength: value.length, options, errors }
+      }))
+    }
+    if (ttl !== undefined && (ttl < 0 || ttl > MAX_TTL_SECONDS)) {
```

Comment:
too bad AJV validator can't have a min max :/

Reply:
yes AJV has a min max, but the error message is not nice... 
```
AioCoreSDKError [AdobeStateLibError]: [AdobeStateLib:ERROR_BAD_ARGUMENT] WARNING: keyword 'minimum' was not handled for formatting: {
  "instancePath": "/ttl",
  "schemaPath": "#/properties/ttl/minimum",
  "keyword": "minimum",
  "params": {
    "comparison": ">=",
    "limit": 0
  },
  "message": "must be >= 0"
}
```

I wanted to make it clear that infinite ttls are not supported anymore!

Reply:
actually we can add the support in our code here, but maybe for next time: https://github.com/adobe/aio-lib-state/blob/94e34ad9f4286de105ecda8fd8d917d1f1102b99/lib/utils.js#L63

Reply:
great maybe next time, as this is a custom error anyways